pkg/elasticsearch/cluster.go

```diff
@@ -222,17 +224,22 @@ func (ec *esClient) GetLowestClusterVersion() (string, error) {
 		return "", payload.Error
 	}
 	if payload.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("failed to get cluster state for %q. Error code: %v, %v", ec.cluster, payload.StatusCode != http.StatusOK, payload.ResponseBody)
+		return "", errors.New("failed to get cluster state",
+			"cluster", ec.cluster,
+			"response_code", payload.StatusCode,
+			"response_body", payload.ResponseBody)
 	}
 
 	res := &estypes.StatsNodesResponse{}
 	err := json.Unmarshal([]byte(payload.RawResponseBody), res)
 	if err != nil {
-		return "", fmt.Errorf("failed decoding raw response body into `estypes.StatsNodesResponse` for %s: %s", ec.cluster, err)
+		return "", errors.Wrap(err, "failed to decode raw response body into `estypes.StatsNodesResponse`",
```

Comment:
This is misleading - standard errors.Wrap() is about saving stack traces which is entirely different from what you are doing. I suggest renaming to `kverror.WithCause(err, nvpairs...) KVError` and `KVError.Cause() error` 
Wrap and Unwrap should keep their standard meanings otherwise KVError is breaking the `error` contract.

Reply:
There is no `errors.Wrap` in the standard library. Are you talking about something else? The standard library uses `fmt.Errorf` with the `%w` placeholder and `errors.Unwrap` to return the wrapped error. Wrap and Unwrap are pretty standard naming conventions across the board and this does not deviate from that behavior in any way that I can tell. Wrap wraps an error with more context. Unwrap removes the wrapped context and returns the error that was wrapped. The naming convention is not only standard, it's imperative to work with the standard library's `Unwrap` functionality. It cannot be changed.

Reply:
https://godoc.org/github.com/pkg/errors#Wrap - it was introduced in go 1.13, see discussion at https://blog.golang.org/go1.13-errors.
I agree with having an Unwrap() method since KVError can act as an error container for its Cause. I can live with kverror.Wrap() now it's in a distinct package, although I prefer WithCause/Cause because the semantics are different. E.g.

Reply:
Okay gotcha. `pkg` is not the standard library. I thought you were talking about https://golang.org/src/errors/ :smile: 

At this point, it's irrelevant since I renamed the package though so I'll mark this as resolved :+1: